PWGHF/HFL/Tasks/taskHFmuCharmBeautySep.cxx

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \Task to estimate HF->mu in the forward direction and use the DCA observable to separate b-> mu, c-> mu.
+/// \Command to run on terminal : o2-analysis-hfmu-charm-beauty-separation | o2-analysis-fwdtrackextension --aod-file AO2D.root
+/// \Author : Shreyasi Acharya <shreyasi.acharya@cern.ch>, LPC, France
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "TDatabasePDG.h"
+#include "PWGDQ/Core/VarManager.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/ASoAHelpers.h"
+#include "PWGDQ/DataModel/ReducedInfoTables.h"
+#include "PWGDQ/Core/HistogramManager.h"
+#include "PWGDQ/Core/MixingHandler.h"
+#include "PWGDQ/Core/AnalysisCut.h"
+#include "PWGDQ/Core/AnalysisCompositeCut.h"
+#include "ReconstructionDataFormats/TrackFwd.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+// using MyMuons = soa::Join<aod::FwdTracks, aod::FwdTracksDCA>;
+// Iterate on muon using the collision iterator in the dq-analysis style
+struct IterateFwdTracksMuons {
+
+  HistogramRegistry spectra{"ForwardTracks", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
```

Comment:
```suggestion
  HistogramRegistry registry{"registry", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
```

Reply:
Is it okay to add : HistogramRegistry spectra{"spectraForwardTracks", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
Since, in the future, I would like to add more HistogramsRegistry ?

Reply:
Ah, I was thinking of Adding separate ones for some QA checks. Is that okay ?

Reply:
Would they have different settings from this one? In case you need a separate category of histograms, why not putting them in a dedicated directory?
Btw, do you need to provide the arguments `OutputObjHandlingPolicy::AnalysisObject, true, true`? Aren't these the default values?

Reply:
I was trying following the structure from : https://github.com/AliceO2Group/O2Physics/blob/master/Tutorials/src/histogramRegistry.cxx (line 136 struct RealisticExample)
But I can change it to HistogramRegistry registry{"registry".... Please let me know.

Reply:
The first two tasks in that tutorial use the simple registry declaration. Please use that one unless you have specific reasons to do otherwise.

Reply:
okie, no problem at the moment. I changed it.